### PR TITLE
[prometheus-blackbox-exporter] fix linting failure due to deprecated api version (see issue #56)

### LINKED ISSUE #56: [prometheus-blackbox-exporter] Linting fails due to use of deprecated apiVersion rbac.authorization.k8s.io/v1beta1
**Describe the bug**

When running `helm lint` for the `prometheus-blackbox-exporter` chart `4.3.1` it fails with the following errors:

```
  ==> Linting /tmp/750049035/monitoring/prometheus-blackbox-exporter/helm-charts-stable/prometheus-blackbox-exporter/4.3.0/prometheus-blackbox-exporter
  [INFO] Chart.yaml: icon is recommended
  [ERROR] templates/role.yaml: the kind "rbac.authorization.k8s.io/v1beta1 Role" is deprecated in favor of "rbac.authorization.k8s.io/v1 Role"
  [ERROR] templates/rolebinding.yaml: the kind "rbac.authorization.k8s.io/v1beta1 RoleBinding" is deprecated in favor of "rbac.authorization.k8s.io/v1 RoleBinding"
  Error: 1 chart(s) linted, 1 chart(s) failed
```

**Version of Helm and Kubernetes**:

Helm `v3.3.1`
Kubernetes `v1.19.0`

**Which chart**:

prometheus-blackbox-exporter

**What happened**:

Fails linting due to a deprecated api version in `role.yaml` and `rolebinding.yaml`

**What you expected to happen**:

Linting to succeed

**How to reproduce it** (as minimally and precisely as possible):

* Clone the repository
* Navigate to directory `charts/prometheus-blackbox-exporter`
* Run Helm (3+) linting with `helm lint`

**Anything else we need to know**:

Nope

### PR DESCRIPTION
**Pull request by JorritSalverda**
Signed-off-by: Jorrit Salverda <jsalverda@travix.com>

#### Is this a new chart

No

#### What this PR does / why we need it:

To fix `helm lint` from failing on deprecated api versions in `role` and `rolebinding`

#### Which issue this PR fixes

  - fixes #56

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-blackbox-exporter/templates/role.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.pspEnabled }}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (monotek) [inline charts/prometheus-blackbox-exporter/templates/role.yaml:2]: Imho this is a breaking change for users running on older k8s clusters.
Should be implemented via semver version check.

File: charts/prometheus-blackbox-exporter/templates/rolebinding.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.pspEnabled }}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (monotek) [inline charts/prometheus-blackbox-exporter/templates/rolebinding.yaml:2]: Imho this is a breaking change for users running on older k8s clusters.
Should be implemented via semver version check.

File: charts/prometheus-blackbox-exporter/templates/role.yaml (hunk 1 above)
Comment (JorritSalverda) [reply charts/prometheus-blackbox-exporter/templates/role.yaml]: Fair enough, will address it through a template in the helper, just like the `prometheus-cloudwatch-exporter` does.

**Comment by JorritSalverda** [comment]
Ouch, can I retroactively sign my second commit?

**Comment by scottrigby** [comment]
`git commit --amend --signoff`

See the DCO check status page for more info 🙂 https://github.com/prometheus-community/helm-charts/runs/1082200117

**Comment by JorritSalverda** [comment]
> `git commit --amend --signoff`
> 
> See the DCO check status page for more info 🙂 https://github.com/prometheus-community/helm-charts/runs/1082200117

Ah, that helps. Super!

**Review by monotek — APPROVED** [review]

**Pull request merged**